ide/csl.api/src/org/netbeans/modules/csl/editor/semantic/SemanticHighlighter.java

```diff
@@ -133,139 +109,39 @@ public final void cancel() {
             }
 
             final GsfSemanticLayer layer = GsfSemanticLayer.getLayer(SemanticHighlighter.class, doc);
-            SwingUtilities.invokeLater(new Runnable () {
-                public void run() {
-    // XXX: parsingapi
-                    layer.setColorings(newColoring, -1); //version
-                }
+            SwingUtilities.invokeLater(() -> {
+                // XXX: parsingapi
+                layer.setColorings(newColoring);
             });
         } finally {
             SpiSupportAccessor.getInstance().removeCancelSupport(cancel);
         }
     }
     
+    private void processColorings(ResultIterator resultIterator, SortedSet<SequenceElement> newColoring) throws ParseException  {
+        String mimeType = resultIterator.getSnapshot().getMimeType();
+        Language language = LanguageRegistry.getInstance().getLanguageByMimeType(mimeType);
+        if (language != null) {
+            ColoringManager manager = language.getColoringManager();
+            SemanticAnalyzer task = language.getSemanticAnalyzer();
+            if (manager != null && task != null) {
+                Parser.Result r = resultIterator.getParserResult();
+                if (r instanceof ParserResult) {
+                    process(language, (ParserResult) r, newColoring);
+                }
+            }
+        }
```

Comment:
I think this can be simplified to (`manager` and `task` can't be null, at least that is my reading from the code and assured by api annotations).

```suggestion
        if (language != null) {
            Parser.Result r = resultIterator.getParserResult();
            if (r instanceof ParserResult parserResult) {
                process(language, parserResult, newColoring);
            }
        }
```